### PR TITLE
[5.1] Adjustment to allow setContentTags to handle escaping of tags

### PR DESCRIPTION
**Pull request by TerrePorter**
updated to use 5.1 version
- removed setEchoFormat function and $echoFormat variable
- adjusted compileRegularEchos function
- adjusted setContentTags function
- adjusted setEscapedContentTags function

I could find no other use in the existing code for the setEchoFormat, so I'm assuming it was in to allow to change to support the old way content tags were being escaped or not.

This fixes the setContentTags to support setting tags to be escaped or not.

In the function setContentTags($openTag, $closeTag, $escaped) the original $escaped flag only changed the tag name, nothing to really do with if the specified tag is escaped or not.

Default in setContentTags function for $escaped is true, this keeps the default of the content tags being escaped.

This change allows the developer to set the standard content tags to be escaped or not and makes the function more in line with how it reads.

Example:
setContentTags('[[', ']]') - would work just like the default tags, they would be escaped
setContentTags('[[', ']]', false) - would set the tags to not be escaped

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Tests are failing.

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 1]
```diff
@@ -800,13 +804,12 @@ protected function getTags($escaped = false)
...
-     */
-    public function setEchoFormat($format)
-    {
-        $this->echoFormat = $format;
+    * Set if content tags should be escaped
+    *
+    * @param  bool  $escaped
+    * @return void
+    */
+    public function setContentTagsEscaped($escaped = true) {
```
Comment (GrahamCampbell) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: cs

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 2]
```diff
@@ -61,11 +61,11 @@ class BladeCompiler extends Compiler implements CompilerInterface
     protected $escapedTags = ['{{{', '}}}'];
 
     /**
-     * The "regular" / legacy echo string format.
+     * To keep track of if content tag are escaped
```
Comment (GrahamCampbell) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: missing fullstop

Comment (GrahamCampbell) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: Also, the English here isn't so good.

**Comment by TerrePorter** [comment]
Ok, I will adjust the code for the cs.

I looked at the test failure and that is because of the removal of the setEchoFormat function. Not sure what I should do about that.

**Comment by GrahamCampbell** [comment]
> Not sure what I should do about that.

Write new tests for your changes (including modifying the existing ones).

**Comment by TerrePorter** [comment]
I finally was able to get a local working copy with the tests running. I updated the test that was failing. 

As far as I can tell, the existing tests are checking the code that has changed. I'm new to phpunit, so I could be wrong.

Thanks for taking the time to look at this.

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 3]
```diff
@@ -800,13 +804,12 @@ protected function getTags($escaped = false)
...
+     * Enable/Disable escape setting for contentTags.
      *
-     * @param  string  $format
+     * @param  bool  $escaped
      * @return void
      */
-    public function setEchoFormat($format)
-    {
-        $this->echoFormat = $format;
+    public function setContentTagsEscaped($escaped = true) {
```
Comment (GrahamCampbell) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: please follow psr-2

**Comment by taylorotwell** [comment]
I can't figure out what is being attempted here.

**Comment by TerrePorter** [comment]
@taylorotwell This proposal fixes the setContentTags to handle if the tags specified for the content tags are or are not escaped. This can be accomplished right now with the function setEchoFormat but it is cumbersome.

To overwrite the default of escaping the content tags, you would do something like this.

```
Blade::setEchoFormat ('%s');
Blade::setContentTags('{{','}}');
```

The change removes the setEchoFormat and allows the escaped or not to be set on the setContentTags function, similar to how it reads.

With the changes, I can do this to set the content tags to not be escaped. It is true by default that keeps the standard of them being escaped.

```
Blade::setContentTags('{{','}}', false);
```

This makes it easier to change the default tags to be or not be escaped. Which is helpful when sending templates to JavaScript frameworks that do not want the content escaped or in my case where I am assembling several small templates to make a bigger template then saving that for use in the cms. So I turn on and off the escaping of the tags at random times, using the preset escaped tags causes other problems for me.

The other think it does it make the function make better sense.

Currently this is the setContentTags function.
```
/**
     * Sets the content tags used for the compiler.
     *
     * @param  string  $openTag
     * @param  string  $closeTag
     * @param  bool    $escaped
     * @return void
     */
   public function setContentTags($openTag, $closeTag, $escaped = false)
    {
        $property = ($escaped === true) ? 'escapedTags' : 'contentTags';

        $this->{$property} = [preg_quote($openTag), preg_quote($closeTag)];
    }
```
To me that reads as if I set the $escaped to true then the tags specified would be escaped, but instead it switches the variable to escapedTags and then sets the tags specified to that variable, the content tags doesn't change.

Why would I use the contentTags to set the escapedTags when there is a escapedTags function to do that?

Which I also adjusted as it was calling the setContentTags function but only with the $escaped = true.

With the changes, the setContentTags now sets the contentTags variable with the option to escape or not escape the tags. The setEscapedContentTags only changes the tags used for the escapedTags.

I hope that explains it better. Perhaps you can reconsider the request.

**Comment by TerrePorter** [comment]
@taylorotwell I added in a explanation as to the reason behind the change I'm assuming that you haven't time to take another look or have I not explained it well enough?

**Comment by TerrePorter** [comment]
@GrahamCampbell, @taylorotwell Any chance of someone taking another look at this?

**Pull request closed without merging**